packages/yoastseo/src/languageProcessing/helpers/word/markWordsInSentences.js

```diff
@@ -76,9 +77,30 @@ const getMarkedAnchors = function( sentence, wordsRegex ) {
  * @returns {string} The sentence with marks.
  */
 export const collectMarkingsInSentence = function( sentence, wordsFoundInSentence, matchWordCustomHelper ) {
-	wordsFoundInSentence = wordsFoundInSentence.map( word => escapeRegExp( word ) );
+	const allWordsFound = [];
+	wordsFoundInSentence.forEach( word => {
+		// Check if the word in `wordsFoundInSentence` contains a single quote.
+		const matchedSingleQuote = word.match( SINGLE_QUOTES_REGEX );
+		if ( matchedSingleQuote ) {
+			/*
+			 * If yes, make all different combinations of the word with different types of single quotes in the array.
+			 * Later, a regex will be created for all words that were found in the sentence including their variations.
+			 *
+			 * For example:
+			 * `wordsFoundInSentence`: [ "red", "panda’s" ]
+			 * For the word "panda’s", we'll create the following variations:
+			 * "panda's", "panda‘s", "panda’s", "panda‛s", "panda`s", "panda‹s", "panda›s"
+			 * And those variations will be added to `allWordsFound`.
+			 */
+			SINGLE_QUOTES_ARRAY.forEach( singleQuote => {
+				allWordsFound.push( escapeRegExp( word.replace( matchedSingleQuote.toString(), singleQuote ) ) );
```

Comment:
The `.toString()` does not work in the off-chance there are multiple single quotes in `word`, in which case `matchedSingleQuote` is an array of multiple strings: then, it would add a `,` between the results (see [this documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toString)). In English there are a few of these words, cf. [this overview](https://en.wiktionary.org/wiki/Category:English_double_contractions), most notable `fo'c's'le`. I'm not sure about other languages, but I would consider looping over the `matchedSingleQuote` variable 😄

Reply:
Thanks Martijn! that's indeed a very good point. Looping over the matched single quotes would be the best indeed 🙌🏽